### PR TITLE
uftrace: Fix .travis.yml to use install-dep.sh

### PR DESCRIPTION
**Pull request by gonapps**
With this change you will immediately know
when change to install-deps.sh introduces some errors especially in Ubuntu.

Signed-off-by: Byeonggon Lee <gonny952@gmail.com>

### CODE REVIEW

**Comment by honggyukim** [comment]
The current setting looks Ubuntu 14.04 LTS, but it doesn't have libcapstone-dev so might be a problem. In addition, you better add `-y` option to `install-deps.sh`.

**Comment by gonapps** [comment]
@honggyukim Thanks, I fixed.

File: misc/install-deps.sh [hunk 1]
```diff
@@ -14,20 +14,20 @@ distro=$(grep "^ID=" /etc/os-release | cut -d\= -f2 | sed -e 's/"//g')
 case $distro in
     "ubuntu" | "debian")
         apt-get $OPT install pandoc libdw-dev libpython2.7-dev libncursesw5-dev pkg-config
-        apt-get $OPT install libcapstone-dev libluajit-5.1-dev ;;
+        apt-get $OPT install libcapstone-dev libluajit-5.1-dev || true ;;
```
Comment (DanielTimLee) [inline misc/install-deps.sh]: Seems you're trying to suppress the error from the absence of the package `libcapstone-dev` which doesn't exist in Ubuntu 14.04 LTS. 

But with this approach, this will skip the installation of `libcapstone-dev` as well as `libluajit-5.1-dev`.

https://pkgs.org/download/libluajit-5.1-dev

Since luajit package exist in Ubuntu 14.04 LTS, how about moving `libluajit-5.1-dev` to upper line?

**Comment by gonapps** [comment]
@DanielTimLee Okay I fixed it.

**Comment by DanielTimLee** [comment]
@gonapps  Thanks for the applying the feedback!
But actually what I meant was, `libluajit-5.1-dev` won't be necessary with the true option,
since the it is already there in the Ubuntu 14.04 package repository.

(I'm not so sure about the other disto)

**Comment by namhyung** [comment]
Actually I'd like to update to Ubuntu 16.04 if possible since 14.04 is in EOL (end-of-life) now.

**Comment by gonapps** [comment]
@namhyung @DanielTimLee I modified travis to use ubuntu 16.04, so.. how do you think about '|| true'? should I remove '|| true' as before modification?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,15 @@
 language: c
 
 # Ubuntu 14.04 LTS
-dist: trusty
+dist: xenial
```
Comment (namhyung) [inline .travis.yml]: Please update the comment too,

File: .travis.yml [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
 sudo: required
 
 install:
 - sudo apt-get -qq update
-- sudo apt-get -y  install libelf-dev libstdc++-4.8-dev pandoc
+- sudo misc/install-deps.sh -y
 
-script: ./configure && make
+script: ./configure && make && tests/unittest
```
Comment (namhyung) [inline .travis.yml]: It won't build the unittest binary so will fail.  I think you can call `make test TESTARG=unittest` instead.

File: misc/install-deps.sh [hunk 4]
```diff
@@ -14,20 +14,25 @@ distro=$(grep "^ID=" /etc/os-release | cut -d\= -f2 | sed -e 's/"//g')
 case $distro in
     "ubuntu" | "debian")
         apt-get $OPT install pandoc libdw-dev libpython2.7-dev libncursesw5-dev pkg-config
-        apt-get $OPT install libcapstone-dev libluajit-5.1-dev ;;
+        apt-get $OPT install libluajit-5.1-dev || true
+        apt-get $OPT install libcapstone-dev || true ;;
```
Comment (namhyung) [inline misc/install-deps.sh]: Is this needed for travis?  Anyway I think this change should be in a separate commit.

File: misc/install-deps.sh [hunk 5]
```diff
@@ -14,20 +14,25 @@ distro=$(grep "^ID=" /etc/os-release | cut -d\= -f2 | sed -e 's/"//g')
...
         yum install $OPT pandoc elfutils-devel python2-devel ncurses-devel pkgconfig
-        yum install $OPT capstone-devel luajit-devel ;;
+        yum install $OPT luajit-devel || true
+        yum install $OPT capstone-devel || true ;;
     "arch" | "manjaro")
-        pacman $OPT -S pandoc libelf python2 ncurses pkgconf
-        pacman $OPT -S capstone luajit ;;
+	pacman $OPT -S pandoc libelf python2 ncurses pkgconf
+	pacman $OPT -S luajit || true
+	pacman $OPT -S capstone || true ;;
```
Comment (namhyung) [inline misc/install-deps.sh]: Please fix the indentation, we used all spaces here.

File: .travis.yml (hunk 2 above)
Comment (gonapps) [reply .travis.yml]: Thanks for the review. I fixed this

File: .travis.yml (hunk 3 above)
Comment (gonapps) [reply .travis.yml]: Oh, my mistake. I changed this to 'make uniitest'

File: misc/install-deps.sh (hunk 4 above)
Comment (gonapps) [reply misc/install-deps.sh]: Because .travis.yml is now using ubuntu 16.04, this change is not much necessary. Should I revert this change as before?

File: misc/install-deps.sh (hunk 5 above)
Comment (gonapps) [reply misc/install-deps.sh]: Thanks. I fixed.

File: misc/install-deps.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+
```
Comment (namhyung) [inline misc/install-deps.sh]: ???

File: misc/install-deps.sh (hunk 4 above)
Comment (namhyung) [reply misc/install-deps.sh]: Well, it's not important.  Just leave it.

File: misc/install-deps.sh (hunk 6 above)
Comment (gonapps) [reply misc/install-deps.sh]: My mistake.. I fixed

**Comment by namhyung** [comment]
Well, I think the most important change here is upgrading Ubuntu version and running unit test.  Could you please update the commit message?

**Comment by gonapps** [comment]
@namhyung Okay I updated the commit message!

**Comment by namhyung** [comment]
The subject line should be shorter than that.  How about "build: Update Travis CI to use Ubuntu 16.04 and unit test"?

**Comment by gonapps** [comment]
@namhyung Updated

File: .travis.yml (hunk 3 above)
Comment (namhyung) [reply .travis.yml]: Oh, you're right
